### PR TITLE
add atbash, crout, hosoya triangle algorithms

### PR DESCRIPTION
**Pull request by Notheryne**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

- [X] **If creating a new file :**
  - [X] added links to it in the README files ?
  - [X] included tests with it ?
  - [X] added description (overview of algorithm, time and space compleixty, and possible edge case) in docstrings ?

- [ ] **if done some changes :**
  - [ ] wrote short description in the PR explaining what the changes do ?
  - [ ] Fixes #[issue number] if related to any issue

- [ ] **other**

### CODE REVIEW

File: algorithms/dp/hosoya_triangle.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+5 3 4 3 5 
+8 5 6 6 5 8
+
+The complexity is O(n^3).
+
+"""
+
+
+
+def Hosoya(n, m):
```
Comment (keon) [inline algorithms/dp/hosoya_triangle.py]: `hosoya`

File: algorithms/dp/hosoya_triangle.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        (n == 1 and m == 1) or (n == 2 and m == 1)): 
+        return 1
+    if n > m: 
+        return Hosoya(n - 1, m) + Hosoya(n - 2, m) 
+    elif m == n: 
+        return Hosoya(n - 1, m - 1) + Hosoya(n - 2, m - 2) 
+    else: 
+        return 0
+          
+def printHosoya(n):
```
Comment (keon) [inline algorithms/dp/hosoya_triangle.py]: print_hosoya

**Comment by Notheryne** [comment]
Ok, I'll work on the tests throughout the weekend, I should be able to include them by sunday.

**Comment by Notheryne** [comment]
I've added the unittests, updated non-english readmes, made some descriptions better and changed crout to always return floats.
I had to make a new function in hosoya for testing, because it is supposed to only print. Hope it's good.

**Comment by keon** [comment]
@Notheryne Thanks for the contribution!

**Pull request merged**